### ISSUE TITLE
Gray-Ham Glee-Sin reporting for duty

### ISSUE THREAD

**Issue by gragleas** [brianfunk3/PIEthon#70]
Forgot to do this on thursday night so here I am!

**Comment by brianfunk3** [comment]
Not sure if you figured out the whole abbreviation issue but it seem like you were on track with the solution we talked about. Let me know if you encounter anything else either here or when we work together

**Comment by gragleas** [comment]
I re-did what I have so far using pandas, joining the semesters dict, and using group by. I'm able to run main and make a plot for each lab (contacts per day for each semester for each lab). The only issue I have is importing the module I made. I can't save it locally on the computers at here (at wells), and I can't mount my google drive I'm assuming because of the duo authentication for the google account. To run main I upload the local module, but that gets deleted when I close the notebook.
[GG_data_team_project.zip](https://github.com/brianfunk3/PIEthon/files/3877990/GG_data_team_project.zip)

**Comment by gragleas** [comment]
Hello! I've started playing around with some data and right now I have the average number of contacts per shift time, per semester in the LC since 2014. The only thing is that the shift start times are all 0-24 which I'm assuming is just midnight to midnight, but some of these numbers are a little sketch. Like I don't think we average 20 contacts a shift for an 8-midnight in the LC over summer. Anyways that's where I'm at, I've been looking around at the z-score method you suggested and I'll keep that in mind. I'll try to clean up those shift times and see where I can go from there.

**Comment by brianfunk3** [comment]
Yeah I'm looking at that now and it definitely is wrong, I think its an issue with timezone converting. I'll try to get it resolved by the end of my shift today, and I'll comment here when it is

**Comment by brianfunk3** [comment]
yup I'm dumb and don't know how timezones work. PIEthon zip in the box has been updated with new version that corrects this, thanks for pointing it out Graham

**Comment by brianfunk3** [comment]
okay actually the change I made broke the parsing of date information so I'll have to take a look at during my next shift. Times are still correct now but post-processing is a lil wonky

**Comment by brianfunk3** [comment]
whoops didn't mean to close

**Comment by gragleas** [comment]
Hello again! I've brute forced my functions to work with a lot of casting, but I've run into another issue. When I'm counting the contacts in the LC per shift time per semester, my counts are significantly lower than they should be. I found that a lot of the shift times are just missing from the PIEthon shifts pull. For example there are a couple thousand contacts in November of 2014, but there are NO Fall 2014 shifts in the pull at all, and since each contact doesn't have a matching shift, it passes straight through my counter without increasing. I've noticed that the numbers were off for some other semesters too, but I haven't done the math out with them. Do you know anything about the said missing semesters?

**Comment by brianfunk3** [comment]
Hmmm that is very interesting. I have no idea what could be going on, but I'll look into it. If anything I'm meeting with the Pie devs on Wednesday and I'll ask what their thoughts on it are
